test/test_response.py

```diff
@@ -198,7 +200,9 @@ def test_io(self):
         self.assertRaises(IOError, resp3.fileno)
 
     def test_io_closed_consistently(self):
-        hlr = httplib.HTTPResponse(socket.socket())
+        sock = socket.socket()
+        self.addCleanup(sock.close)
```

Comment:
Reviewing the [docs on socket programming](https://docs.python.org/3.5/howto/sockets.html#disconnecting):

> Python takes the automatic shutdown a step further, and says that when a socket is garbage collected, it will automatically do a `close` if it’s needed. But relying on this is a very bad habit. If your socket just disappears without doing a `close`, the socket at the other end may hang indefinitely, thinking you’re just being slow. _Please_ `close` your sockets when you’re done.

So maybe have an explicit `close` on both resources? Would that be acceptable @Lukasa?